### PR TITLE
Woop woop rework

### PR DESCRIPTION
**Pull request by varholak-peter**

### CODE REVIEW

File: src/app/app.js [hunk 1]
```diff
@@ -12,11 +13,11 @@ class App extends React.Component<{}> { // eslint-disable-line react/prefer-stat
   render() {
     return (
       <Store>
-        <Router>
+        <Router history={history}>
           <div>
             <Header />
             <div className="content">
-              <BackButton />
+              <Route path="/(.+)" component={BackButton} />
```
Comment (garySiladi) [inline src/app/app.js]: whats this syntax?

File: src/components/analysis-page/analysis-header.js [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
+  +actualResult: Array<Number>,
+};
+
+const AnalysisHeaderView = ({ previewImage, expectedResult, actualResult }: Props) => (
   <div className="analysis-header">
-    <div>Found tumors</div>
-    <div>Expected Result</div>
-    <div>Calculated Result</div>
     <div>
+      <img src={previewImage} alt="..." className="analysis-image" />
```
Comment (garySiladi) [inline src/components/analysis-page/analysis-header.js]: maybe we could pass the `name` and use that for alt

File: src/components/analysis-page/analysis-result.css [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+.result-number {
+  font-size: 2em;
+}
+
+.result-alert {
+  width: 1.5em;
+  height: 1.5em;
+  margin-right: 1em;
+  margin-left: -2.5em;
```
Comment (garySiladi) [inline src/components/analysis-page/analysis-result.css]: although I don't like the neg. value, as I looked at the wireframes I can understand why you use it 😄

File: src/components/header/header.js [hunk 4]
```diff
@@ -11,7 +11,12 @@ const Header = () => (
       <img src={profilePic} alt="..." className="profile-pic" />
       <Icon name="dropdownArrow" />
     </div>
-    <Icon name="logout" className="logout" />
+    <a href="https://www.google.com/">
```
Comment (garySiladi) [inline src/components/header/header.js]: so this is what you mean by functioning logout 😄

File: src/components/study-collection/study-collection.js [hunk 5]
```diff
@@ -1,51 +1,51 @@
...
+  actions: {
+    dispatchSelectStudy,
   },
-];
+}: Props) => {
+  const renderedStudies = studies.filter(study => study.patientId === selectedPatient.id).map(
+    study => (
+      <StudyCard
+        cardData={study}
+        onSelect={() => dispatchSelectStudy(study)}
```
Comment (garySiladi) [inline src/components/study-collection/study-collection.js]: is this best practice? to dispatch as you redirect in router?

File: src/store/store.js [hunk 6]
```diff
@@ -2,19 +2,29 @@
 import React from 'react';
 import type { Node } from 'react';
 import { Provider } from 'react-redux';
-import { createStore } from 'redux';
+import { createStore, applyMiddleware } from 'redux';
+import * as storage from 'redux-storage';
```
Comment (garySiladi) [inline src/store/store.js]: was this package better than `redux-persist` which I linked you?

File: src/app/app.js (hunk 1 above)
Comment (varholak-peter) [reply src/app/app.js]: *magic* (path works very regex-y so this shows the button on anything that isn't "/")

File: src/components/analysis-page/analysis-header.js (hunk 2 above)
Comment (varholak-peter) [reply src/components/analysis-page/analysis-header.js]: alts are not important now

File: src/components/analysis-page/analysis-result.css (hunk 3 above)
Comment (varholak-peter) [reply src/components/analysis-page/analysis-result.css]: it is bullshit

File: src/components/header/header.js (hunk 4 above)
Comment (varholak-peter) [reply src/components/header/header.js]: I wanted to close the window but JS can't close tabs which were not opened by js ¯\_(ツ)_/¯

File: src/components/study-collection/study-collection.js (hunk 5 above)
Comment (varholak-peter) [reply src/components/study-collection/study-collection.js]: Don't ask me, that's your code: https://github.com/varholak-peter/mdp/blob/master/src/components/table/table-viewer.js#L55

File: src/store/store.js (hunk 6 above)
Comment (varholak-peter) [reply src/store/store.js]: It requires less config, that one saves periodically and you have to add middleware into your reducers if you want to save on specific actions (better for websockets where there are multiple actions per second) this one automatically saves on every action.

File: src/components/study-collection/study-collection.js (hunk 5 above)
Comment (garySiladi) [reply src/components/study-collection/study-collection.js]: As the mighty Rolo always says: "I was just testing you"

**Pull request merged**